packages/core/src/server/options.ts

```diff
@@ -7,7 +7,11 @@ export const apiOptions: APIOptions = {
   account: storeConfig.api.storeId,
   environment: storeConfig.api.environment as APIOptions['environment'],
   subDomainPrefix: storeConfig.api.subDomainPrefix ?? ['www'],
-  hideUnavailableItems: storeConfig.api.hideUnavailableItems,
+  searchOptions: {
+    hideUnavailableItems: storeConfig.api.hideUnavailableItems,
+    simulationBehavior: (storeConfig.api as Record<string, any>)
+      .simulationBehavior,
+  },
```

Comment:
**issue (blocking)**: this type change lead to build errors on `starter.store`:

![Screenshot 2024-09-09 at 10 37 03](https://github.com/user-attachments/assets/1cc05e25-a3bd-4058-9359-5334f7cd10c4)

To ensure backward compatibility we should just add the new attribute and maybe update the `faststore.config.default` to have the updated type by default, what do you think?

Reply:
Good catch, the contract change will be a breaking change.

Reply:
Thanks! Could you test it again, please?